src/annotation.js

```diff
@@ -156,7 +156,7 @@ export class Annotation extends Modifier {
 
     // We're changing context parameters. Save current state.
     this.context.save();
-    const classString = Object.keys(this.getAttribute('classes')).join(' ');
+    const classString = Object.keys(this.classes).join(' ');
```

Comment:
Is this intentional? Does this work on SVG elements?

Reply:
This is one refactoring that I propose in this PR: `this.attrs` 

It is intentional but I do not know if it works with SVG, how can I check that?

Reasons for the change:
- `this.attrs` contains three different types of information: string (`id`, `type`, ...), SVGSVGElement (`el`), Record<string, boolean> (`classes`)
- `el`  is not used in the software, it is only initialised to null
- there are two methods to access attributes `getAttribute` and `setAttribute`
- there are three methods to access classes `hasClass`, `addClass`, `removeClass`

My proposal in this PR:
- Remove `el`
- Leave string information in `this.attr` with its associated acces functions `getAttribute` and `setAttribute`.
- Place the classes in `this.attr` with its associated access functions `hasClass`, `addClass`, `removeClass`.
- The objects that require the complete list of classes are `annotation`, `chordsymbol` and `ornament` and they can use the protected variable `this.classes` directly.

Reply:
@sschmidTU do you have an oppinion on this one? How is that related to SVG Elements in OSMD?

Reply:
i'm not sure, OSMD doesn't use the Annotation class. If you haven't verified that this works on SVG elements, i guess it shouldn't be changed. Otherwise the `classes` change looks fine to me on a first look.

Reply:
> Thanks for the explanation Rodrigo. I agree with @wassertim that there's too much logic refactoring going on here. I think we should separate it out and focus on TypeScript.

@0xfe my focus is on TypeScript I think that we have to as far as possible avoid the usage of `any` in order to get the advantages of TypeScript type checking and and to double types in order to simplify the code.

The explanation is long, but the change related to this is minor this why I put it in this CR to avoid two `any`. Here is the related change, if you still want to remove it I will take it out. :)

In `constructor`:
```diff
     this.attrs = {
      ...
-      classes: {},
     };
+    this.classes = {};
```
In `hasClass`:
```diff
-    return this.attrs.classes[className] === true;
+    return this.classes[className] === true;
```
In `addClass`
```diff
-    this.attrs.classes[className] = true;
+    this.classes[className] = true;
```
In `removeClass`:
```diff
-    delete this.attrs.classes[className];
+    delete this.classes[className];
```

Reply:
I guess I don't understand why we need to change the interface (the `getAttribute` call) to avoid `any`. Either way, attributes are generic key/value pairs, and it's one of the few places `any` is useful, because it allows clients to attach arbitrary objects to elements. So I think (as an exception), `any` is useful here.

Reply:
@0xfe I believe this was an easy instance to avoid `any` but accepted, my biggest concern now is your question _Does this work on SVG elements?_ I will open an issue on this one. 

Just to finish this, do you want to leave the access to the protected variable through the function `getAttribute`:
```js
this.getAttribute('classes')
```
or directly (my preference with protected variables)
```js
this.attr.classes
```

Reply:
`getAttribute` is part of the API, so I would like all accesses to go through there. Part of the idea was also that classes could be inherited (it's not implemented yet), but `getAttribute` lets me change the lookup logic more consistently.

Reply:
`Annotation` is a child of `Element` and has access to its protected function. Just to be sure I leave `attr` `protected` in `Element`, no?

Reply:
I think that this would also break `registry.js` which relies on `getAttribute(...)` to build indexes.

Reply:
Please let's keep `getAttribute` -- there's a lot of things hanging on this :-)

Reply:
@0xfe this is understood my question is: do you want to change `attr` to `private` in the base class (Element) or we leave it `protected` like all the others.

Reply:
Ah, gotcha. Yes, `private` please, so we force all accesses through `getAttribute`.